internal/provider/datasources/data_source_api_tokens_test.go

```diff
@@ -0,0 +1,208 @@
+package datasources_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/astronomer/terraform-provider-astro/internal/clients/iam"
+
+	astronomerprovider "github.com/astronomer/terraform-provider-astro/internal/provider"
+	"github.com/astronomer/terraform-provider-astro/internal/utils"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+)
+
+type checkApiTokensInput struct {
+	filterWorkspaceId  bool
+	filterDeploymentId bool
+	filterOrgOnly      bool
+	workspaceId        string
+	deploymentId       string
+	organizationId     string
```

Comment:
the top 3 are not needed
it makes more sense to check `len(workspaceId)` for `filterWorkspaceId` being true

Reply:
makes sense